src/DataValues/DecimalValue.php

```diff
@@ -93,33 +85,25 @@ private function assertNumberString( $number ) {
 	 * @throws InvalidArgumentException
 	 */
 	private function convertToDecimal( $number ) {
-		if ( !is_int( $number ) && !is_float( $number ) ) {
-			throw new InvalidArgumentException( '$number must be an int or float.' );
-		}
-
 		if ( $number === NAN || abs( $number ) === INF ) {
 			throw new InvalidArgumentException( '$number must not be NAN or INF.' );
 		}
 
 		if ( is_int( $number ) || ( $number === floor( $number ) ) ) {
 			$decimal = strval( abs( (int)$number ) );
 		} else {
-			$decimal = trim( number_format( abs( $number ), 100, '.', '' ), 0 );
+			$decimal = trim( number_format( abs( $number ), 100, '.', '' ), '0' );
 
 			if ( $decimal[0] === '.' ) {
 				$decimal = '0' . $decimal;
 			}
 
-			$last = strlen($decimal)-1;
-
-			if ( $decimal[$last] === '.' ) {
-				$decimal = $decimal . '0';
+			if ( substr( $decimal, -1 ) === '.' ) {
```

Comment:
We just talked about this principle for an hour. This is not unrelated. This is the same class. Boy scout rule. Cleanup code you find hard to read while you are working on closely related code. I will not stop doing this, and I find it more and more frustrating to defeat my position.